RecoTauTag/RecoTau/src/AntiElectronIDMVA6.cc

```diff
@@ -0,0 +1,965 @@
+      } 
+      else {
+        GammasdEtaOutSigCone.push_back(gamma->eta() - thePFTau.eta());
+        GammasdPhiOutSigCone.push_back(gamma->phi() - thePFTau.phi());
+      }
+      GammasPtOutSigCone.push_back(gamma->pt());
+    }
+  }
+  
+  for ( unsigned i = 0 ; i < thePFTau.signalPFChargedHadrCands().size(); ++i ) {
+    reco::PFCandidatePtr charged = thePFTau.signalPFChargedHadrCands().at(i);
+    float dR = deltaR(charged->p4(), thePFTau.leadPFChargedHadrCand()->p4());
+  
+    // charged particles inside the tau signal cone
+    if (dR < std::max(0.05, std::min(0.10, 3.0/thePFTau.pt()))) {
+        pfChargedSum += charged->p4();
+    }
+  }
+  
+  Int_t TauSignalPFGammaCandsIn = GammasPtInSigCone.size();
+  Int_t TauSignalPFGammaCandsOut = GammasPtOutSigCone.size();
+  Float_t TauVisMassIn = (pfGammaSum + pfChargedSum).mass();
+
+  Float_t TauPhi = thePFTau.phi();
+  float sumPhiTimesEnergy = 0.;
+  float sumEnergyPhi = 0.;
+  for ( std::vector<reco::PFCandidatePtr>::const_iterator pfCandidate = signalPFCands.begin();
+	pfCandidate != signalPFCands.end(); ++pfCandidate ) {
+    sumPhiTimesEnergy += (*pfCandidate)->positionAtECALEntrance().phi()*(*pfCandidate)->energy();
+    sumEnergyPhi += (*pfCandidate)->energy();
+  }
+  if ( sumEnergyPhi > 0. ) {
+    TauPhi = sumPhiTimesEnergy/sumEnergyPhi;
+  }
+  Float_t TaudCrackPhi = dCrackPhi(TauPhi, TauEtaAtEcalEntrance);
+  Float_t TaudCrackEta = dCrackEta(TauEtaAtEcalEntrance);
+  Float_t TauHasGsf = thePFTau.leadPFChargedHadrCand()->gsfTrackRef().isNonnull();
+
+  
+  // === electron variables ===
+  Float_t dummyElecEta = 9.9;
+
+  return MVAValue(TauPt,
+                  TauEtaAtEcalEntrance,
+                  TauPhi,
+                  TauLeadChargedPFCandPt,
+                  TauLeadChargedPFCandEtaAtEcalEntrance,
+                  TauEmFraction,
+                  TauLeadPFChargedHadrHoP,
+                  TauLeadPFChargedHadrEoP,
+                  TauVisMassIn,
+                  TaudCrackEta,
+                  TaudCrackPhi,
+                  TauHasGsf,
+                  TauSignalPFGammaCandsIn,
+                  TauSignalPFGammaCandsOut,
+                  GammasdEtaInSigCone,
+                  GammasdPhiInSigCone,
+                  GammasPtInSigCone,
+                  GammasdEtaOutSigCone,
+                  GammasdPhiOutSigCone,
+                  GammasPtOutSigCone,
+                  dummyElecEta,
+                  0.,
+                  0.,
+                  0.,
+                  0.,
+                  0.,
+                  0.,
+                  0.,
+                  0.,
+                  0.,
+                  0.,
+                  0.,
+                  0.,
+                  0.,
+                  0.,
+                  0.,
+                  0.);
+}
+
+double AntiElectronIDMVA6::minimum(double a, double b)
+{
+  if ( std::abs(b) < std::abs(a) ) return b;
+  else return a;
+}
+
+
+#include<array>
```

Comment:
Why is this #include here? It should be at the top of the file with the rest so readers can easily find it.